run-tests.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python
+
+import sys
+import os
+import argparse
+import subprocess
+
+BASEDIR = os.path.abspath(os.path.dirname(__file__))
+
+_formula = "ckan"
+_images = ["centos6"]
+
+
+def get_tag(image, salt=False):
+    tag = "{0}-formula:{1}".format(_formula, image)
+    if salt:
+        return tag + "_salted"
+    else:
+        return tag
+
+
+def image_exists(image):
+    # docker image accept tags only in version >= 1.8...
+    # so workaround with docker history
+    rc = subprocess.call("docker history -q {0} > /dev/null".format(image), shell=True)
+    if rc == 0:
+        return True
+    elif rc == 1:
+        return False
+    else:
+        raise RuntimeError("Cannot test if image exists")
+
+
+def _build(image, salt=False):
+    dockerfile = "test/{0}.Dockerfile".format(image)
+    tag = get_tag(image, salt)
+    if salt:
+        dockerfile_content = open(dockerfile, "rb").read()
+        dockerfile_content += (
+            b"\n"
```

Comment:
Yg Versil a écrit :
> In run-tests.py
> <https://github.com/logilab/ckan-formula/pull/6#discussion_r64340699>:
>
>> +    rc = subprocess.call("docker history -q {0} > /dev/null".format(image), shell=True)
>> +    if rc == 0:
>> +        return True
>> +    elif rc == 1:
>> +        return False
>> +    else:
>> +        raise RuntimeError("Cannot test if image exists")
>> +
>> +
>> +def _build(image, salt=False):
>> +    dockerfile = "test/{0}.Dockerfile".format(image)
>> +    tag = get_tag(image, salt)
>> +    if salt:
>> +        dockerfile_content = open(dockerfile, "rb").read()
>> +        dockerfile_content += (
>> +            b"\n"
>
> Couldn't this whole string be enclosed within a 3-double-quote delimiter
> (|"""|) ?
>


It could but this would look ugly within the function because of left 
dedent.

BTW, I just copied this code from logilab/saemref-formula.